compiler/rustc_trait_selection/src/solve/eval_ctxt/select.rs

```diff
@@ -58,6 +58,12 @@ impl<'tcx> inspect::ProofTreeVisitor<'tcx> for Select {
             )));
         }
 
+        // Don't winnow until `Certainty::Yes` -- we don't need to winnow until
+        // codegen, technically, and all goals should hold in codegen.
+        if matches!(goal.result().unwrap(), Certainty::Maybe(..)) {
+            return ControlFlow::Break(Ok(None));
+        }
```

Comment:
🤔 a goal can be yes while some its candidates are not, please keep the check that `other` is `Yes` as well